### ISSUE TITLE
QuTiP 4.7.4: incompatibility with the latest scipy (1.12.0)

### ISSUE THREAD

**Issue by knesterov** [qutip/qutip#2300]
https://github.com/qutip/qutip/blob/f5149616a4071a273e7e48a63d956836739c4569/qutip/parallel.py#L7

When the latest scipy version is used (1.12.0), QuTiP (4.7.4) cannot be imported since `from scipy import array` is no longer supported in scipy.

Code to reproduce the bug: 
`import qutip`

Output:
```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[1], line 1
----> 1 import qutip

File ~/anaconda3/envs/test/lib/python3.11/site-packages/qutip/__init__.py:133
    131 # graphics
    132 from qutip.bloch import *
--> 133 from qutip.visualization import *
    134 from qutip.orbital import *
    135 from qutip.bloch3d import *

File ~/anaconda3/envs/test/lib/python3.11/site-packages/qutip/visualization.py:24
     22 from qutip.qobj import Qobj, isket
     23 from qutip.states import ket2dm
---> 24 from qutip.wigner import wigner
     25 from qutip.tensor import tensor
     26 from qutip.matplotlib_utilities import complex_phase_cmap

File ~/anaconda3/envs/test/lib/python3.11/site-packages/qutip/wigner.py:19
     17 import qutip
     18 from qutip import Qobj, ket2dm, jmat
---> 19 from qutip.parallel import parfor
     20 from qutip.cy.sparse_utils import _csr_get_diag
     21 from qutip.sparse import eigh

File ~/anaconda3/envs/test/lib/python3.11/site-packages/qutip/parallel.py:7
      1 """
      2 This function provides functions for parallel execution of loops and function
      3 mappings, using the builtin Python module multiprocessing.
      4 """
      5 __all__ = ['parfor', 'parallel_map', 'serial_map']
----> 7 from scipy import array
      8 import multiprocessing
      9 from functools import partial

ImportError: cannot import name 'array' from 'scipy' (/Users/konstantin/anaconda3/envs/test/lib/python3.11/site-packages/scipy/__init__.py)
```

**Comment by evmckinney9** [comment]
The deprecation warning from scipy==1.11 reads
`DeprecationWarning: scipy.array is deprecated and will be removed in SciPy 2.0.0, use numpy.array instead`.

However, clearly this functionality broke before that release; with some discussion here: 
- https://github.com/scipy/scipy/issues/19065#issuecomment-1678439743
- https://github.com/scipy/scipy/pull/19067

**Comment by Ericgig** [comment]
Should be fixed with the 4.7.5 release.
